### PR TITLE
Routed subinterface enhancements

### PR DESCRIPTION
**Pull request by preetham-singh**
#### What I did
Routed subinterfae enhancements HLD [#833](https://github.com/Azure/SONiC/pull/833)
Added support for configuring routed subinterface in short name and long name format
Updated show command to display user configured subinterfaces in correct format.

**Config commands:**
config subinterface (add | del) <subinterface_name> [vlan <1-4094>]

Ex:
config subinterface add Ethernet0.100
config subinterface add Eth64.100 100

**Show commands:**
show subinterfaces status

#### How to verify it
Configure routed subinterface on physical & port channel interfaces.

#### Dependency
This PR is dependent on sonic-buildimage [8761](https://github.com/Azure/sonic-buildimage/pull/8761)

### CODE REVIEW

**Comment by prsunny** [comment]
Can you add unit test?

File: scripts/intfutil [hunk 1]
```diff
@@ -313,12 +314,12 @@ def appl_db_portchannel_status_get(appl_db, config_db, po_name, status_type, por
...
     sub_intf_sep_idx = sub_intf_name.find(VLAN_SUB_INTERFACE_SEPARATOR)
     if sub_intf_sep_idx != -1:
-        parent_port_name = sub_intf_name[:sub_intf_sep_idx]
-        vlan_id = sub_intf_name[sub_intf_sep_idx + 1:]
+        parent_port_name = intf_get_longname(sub_intf_name[:sub_intf_sep_idx])
 
         full_intf_table_name = "INTF_TABLE" + ":" + sub_intf_name
 
         if status_type == "vlan":
+            vlan_id = appl_db.get(appl_db.APPL_DB, full_intf_table_name, status_type)
```
Comment (venkatmahalingam) [inline scripts/intfutil:322]: Where can we find DB migrator change to populate vlan_id field from sub-interface name?

File: config/main.py [hunk 2]
```diff
@@ -5846,6 +5846,123 @@ def smoothing_interval(interval, rates_type):
...
+    ctx.obj = {'db': config_db}
+
+def subintf_vlan_check(config_db, parent_intf, vlan):
+    subintf_db = config_db.get_table('VLAN_SUB_INTERFACE')
+    subintf_names = [k for k in subintf_db if type(k) != tuple]
+    for subintf in subintf_names:
+        sub_intf_sep_idx = subintf.find(VLAN_SUB_INTERFACE_SEPARATOR)
+        if sub_intf_sep_idx == -1:
+            continue
+        if parent_intf == subintf[:sub_intf_sep_idx]:
```
Comment (venkatmahalingam) [inline config/main.py:5871]: Can we please put these new Click commands in this PR description for easy reference? also, hope you are planning to update the Click config guide?

File: scripts/intfutil (hunk 1 above)
Comment (preetham-singh) [reply scripts/intfutil:322]: As per updated design, intfmgrd supports both short name and long name format subinterfaces.
In long name format, vlan_id is derived from subinterface index by intfmgrd.
In short name format, vlan_id is mandatory parameter to be configured by user(as part of click cli).
With this design we do not need db_migrator changes.

File: config/main.py (hunk 2 above)
Comment (preetham-singh) [reply config/main.py:5871]: Done.

**Review by venkatmahalingam — DISMISSED** [review]

**Comment by prsunny** [comment]
@preetham-singh , can you check if the failure is related to the PR?

**Comment by hasan-brcm** [comment]
> 
> 
> @preetham-singh , can you check if the failure is related to the PR?

Hi @prsunny, this PR is dependent on [PR8761](https://github.com/Azure/sonic-buildimage/pull/8761).

**Review by venkatmahalingam — APPROVED** [review]

**Review by prsunny — APPROVED** [review]

**Pull request merged**